### ISSUE TITLE
Feature Request: query_table support on dashboard

### ISSUE THREAD

**Issue by endzyme** [terraform-providers/terraform-provider-datadog#292]
Currently it doesn't appear I can implement the following dashboard json in the terraform provider. Can someone point me in the right direction if I'm using the `datadog_dashboard` resource incorrectly or if I should be seeking help elsewhere. 

Thanks in advance!

Example Dashboard Config:
```json
{
    "title": "Some Timeboard",
    "description": "",
    "widgets": [
        {
            "id": 1,
            "definition": {
                "type": "query_table",
                "requests": [
                    {
                        "q": "max:kubernetes.memory.usage{$cluster,$daemonset,$deployment} by {kube_container_name,some_cluster}",
                        "aggregator": "max",
                        "limit": 10,
                        "order": "desc",
                        "alias": "Max Mem Usage"
                    },
                    {
                        "q": "max:kubernetes.cpu.usage.total{$cluster,$daemonset,$deployment} by {kube_container_name,some_cluster}",
                        "aggregator": "max",
                        "alias": "Max CPU Usage"
                    }
                ],
                "title": "Max Resources"
            }
        },
        {
            "id": 2,
            "definition": {
                "type": "query_table",
                "requests": [
                    {
                        "q": "avg:kubernetes.memory.usage{$cluster,$daemonset,$deployment} by {kube_container_name,some_cluster}",
                        "aggregator": "avg",
                        "limit": 10,
                        "order": "desc",
                        "alias": "Average Mem Usage"
                    },
                    {
                        "q": "avg:kubernetes.cpu.usage.total{$cluster,$daemonset,$deployment} by {kube_container_name,some_cluster}",
                        "aggregator": "avg",
                        "alias": "Average CPU Usage"
                    }
                ],
                "title": "Average Resources"
            }
        }
    ],
    "template_variables": [
        {
            "name": "cluster",
            "default": "*",
            "prefix": "cluster_name"
        },
        {
            "name": "daemonset",
            "default": "*",
            "prefix": "kube_daemon_set"
        },
        {
            "name": "deployment",
            "default": "*",
            "prefix": "kube_deployment"
        }
    ],
    "layout_type": "ordered",
    "is_read_only": false,
    "notify_list": [],
    "dashboard_id": "blorp"
}
```

### Terraform Version: v0.11.14
Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please upgrade because your issue may have already been fixed.

### Affected Resource(s)
Please list the resources as a list, for example:
- datadog_dashboard

**Comment by nmuesch** [comment]
Hey @endzyme , thanks for the report. Currently this isn't an available feature.  I don't have a concrete timeframe on when this would be worked on, but we're always happy to review pull requests in the meantime!

**Comment by rmax** [comment]
@nmuesch I'm new to `terraform`,  what changes would be needed to support this feature?

**Comment by nuclearnic** [comment]
@endzyme @rmax I'm in the process of opening a pull request for this [here](https://github.com/terraform-providers/terraform-provider-datadog/pull/373). Just have to figure out how the acceptance tests work

**Comment by nmuesch** [comment]
Hey, apologies I had missed the previous message. Most new features require them to be implemented in the underlying zorkian [go-datadog-api](https://github.com/zorkian/go-datadog-api) client, then the added terraform feature here. 

@nuclearnic Terraform has some docs about their acceptance tests here https://www.terraform.io/docs/extend/testing/acceptance-tests/teststep.html if you haven't already seen it 😄 And thanks for looking into this !

**Comment by nuclearnic** [comment]
@nmuesch with the holidays happening I missed your message. I actually just [opened the PR](https://github.com/terraform-providers/terraform-provider-datadog/pull/373) and then came and looked here. 

Should I close it and rather see how it fits into (go-datadog-api)[https://github.com/zorkian/go-datadog-api] ? I haven't looked at that yet.

**Comment by nmuesch** [comment]
No worries @nuclearnic. Usually anything you need to change in the vendor folder would need to merged upstream as opposed to changing it directly in this provider as we simply vendor a version of these dependencies.

I took a quick look at your PR, thanks for contributing!,  and it looks like the only changes were comments - https://github.com/terraform-providers/terraform-provider-datadog/pull/373/files#diff-f7000aa1ad67917131b6a863bebc3f10 We'd want to remove those manual changes from here and get them merged upstream. (i.e. here - https://github.com/zorkian/go-datadog-api/blob/master/screen_widgets.go)

I'll try to take a more thorough look at your PR, but from a quick look, seems like you've got the right approach.

**Comment by nuclearnic** [comment]
@nmuesch I removed the comments from the PR so long.

**Comment by nuclearnic** [comment]
@nmuesch I made a PR adding those comments as requested [here](https://github.com/zorkian/go-datadog-api/pull/297).

Could you let me know if there's anything else outstanding for this to go in ? Thanks!

**Comment by nuclearnic** [comment]
@nmuesch I made a PR adding those comments as requested (here)[https://github.com/zorkian/go-datadog-api/pull/297].

Could you let me know if there's anything else outstanding for this to go in ? Thanks!

**Comment by nuclearnic** [comment]
@enbashi I've fixed the formatting in the PR as requested

**Comment by enbashi** [comment]
Approved.

**Comment by nuclearnic** [comment]
> Approved.

@enbashi thanks! At the risk of being annoying, when/how does merging happen ? 🙈

We'd like to start Terraforming all our dashboards and some projects rely quite heavily on query tables.

**Comment by bkabrda** [comment]
Implemented in #373, closing this issue now.
